components/supervisor/pkg/supervisor/tasks.go

```diff
@@ -275,6 +275,7 @@ func (tm *tasksManager) Run(ctx context.Context, wg *sync.WaitGroup) {
 	for _, task := range tm.tasks {
 		select {
 		case <-ctx.Done():
+			successChan <- false
```

Comment:
There already is a success variable that's used in line 282. Why not set `success = false` and break? This way also the reporter in line 288 would be served.